Google.PowerShell/BigQuery/BqDataset.cs

```diff
@@ -242,16 +269,83 @@ public class SetBqDataset : BqCmdlet
         public Dataset Dataset { get; set; }
 
+        /// <summary>
+        /// <para type="description">
+        /// Sets the labels in Keys to the values in Values for the target Dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.SetLabel)]
+        public SwitchParameter SetLabel { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Clears the keys in Keys for the target Dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.ClearLabel)]
+        public SwitchParameter ClearLabel { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Comma separated list of keys to set or clear from a dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.SetLabel)]
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.ClearLabel)]
+        [ValidateNotNullOrEmpty]
+        public string[] Keys { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// List of values corresponding to the list of keys in -Keys.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, Position = 1, ParameterSetName = ParameterSetNames.SetLabel)]
+        public string[] Values { get; set; }
+
         protected override void ProcessRecord()
         {
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.Default:
+                    break; //No processing needs to be done here
+                case ParameterSetNames.SetLabel:
+                    if (Keys.Length != Values.Length)
+                    {
+                        ThrowTerminatingError(new ErrorRecord(new Exception("Keys and Values must be of the same length"),
+                            $"Keys is of length {Keys.Length}, but Values is of length {Values.Length}.",
+                            ErrorCategory.InvalidArgument,
+                            Keys));
+                    }
+                    Dataset.Labels = Dataset.Labels ?? new Dictionary<string, string>();
+                    for (int i = 0; i < Keys.Length; i++)
+                    {
+                        Dataset.Labels.Remove(Keys[i]);
+                        Dataset.Labels.Add(Keys[i], Values[i]);
+                    }
+                    break;
+                case ParameterSetNames.ClearLabel:
+                    foreach (string key in Keys)
+                    {
+                        Dataset.Labels.Remove(key);
+                    }
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
             Dataset response;
-            bool needToInsert = false;
+            Dataset.ETag = null;
```

Comment:
```
Set-BqDataset : Google.Apis.Requests.RequestError
Resource gcloud-powershell-testing:test_label_1 did not meet condition IF_MATCH. This may be caused by concurrent
updates to a resource. [412]
Errors [
    Message[Resource gcloud-powershell-testing:test_label_1 did not meet condition IF_MATCH. This may be caused by
concurrent updates to a resource.] Location[If-Match - header] Reason[preconditionFailed] Domain[global]
]
At line:1 char:17
+ $data = $data | Set-BqDataset -SetLabel @{"test"="one"}
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Set-BqDataset], GoogleApiException
    + FullyQualifiedErrorId : Google.GoogleApiException,Google.PowerShell.BigQuery.SetBqDataset

```